libraries/SITL/SIM_GPS.cpp

```diff
@@ -1264,4 +1513,14 @@ GPS::gps_data GPS::interpolate_data(const gps_data &d, uint32_t delay_ms)
     return _gps_history[N-1];
 }
 
+float GPS::heading(const gps_data *d) const
+{
+    return wrap_360(ToDeg(atan2f(d->speedE, d->speedN)));
+}
+
+float GPS::speed_2d(const gps_data *d) const
+{
+    return norm(d->speedN, d->speedE);
+}
```

Comment:
Make these methods, and use `wrap_360(ToDeg(Vector2f{d->speedE, d->speedN}.angle()))`

... and we we really need wrap_360?!

Reply:
Split off here: https://github.com/ArduPilot/ardupilot/pull/24525